lib/pbench/agent/tool_meister.py

```diff
@@ -1518,6 +1539,36 @@ def end_tools(self, data):
                     name,
                 )
                 failures += 1
+
+        # Remove persistent tool temporary working directory
+        directory = data["directory"]
+        tool_dir = self.directories[directory]
+
+        self.logger.debug(
+            "%s: deleting persistent tool tmp directory %s %s",
+            self._hostname,
+            self._group,
+            tool_dir,
+        )
+        unexpected_files = []
+        for dir, _, files in os.walk(tool_dir):
+            dirpath = Path(dir).relative_to(tool_dir)
+            if files:
+                unexpected_files += map(lambda x: f"{dirpath}/{x}", files)
+
+        if unexpected_files:
+            self.logger.warning(
+                "%s: unexpected temp files %s",
+                self._hostname,
+                ",".join(unexpected_files),
```

Comment:
While Peter's comment was technically valid, pragmatically we're unlikely to ever want to disable `warning` output ... and even if we did, we're still computing the `",".join` here anyway, which isn't so easy to avoid (if we really care, this would have to be transformed into an `if self.logger.isEnabledFor(logging.WARNING)` block, in which case the f-string would probably be at least as good). Still, this looks a bit more consistent than the f-string form, so that's good.

Reply:
The typical reason I make this comments is more for the discipline, since `.debug` and `.info` statements can become quite costly performance wise as a result of a lapse.